Mathlib/Data/List/Sort.lean

```diff
@@ -378,210 +377,89 @@ end Correctness
 
-@[simp]
-theorem mergeSort'_nil : [].mergeSort' r = [] := by rw [List.mergeSort']
-
-@[simp]
-theorem mergeSort'_singleton (a : α) : [a].mergeSort' r = [a] := by rw [List.mergeSort']
-
-theorem map_merge (f : α → β) (r : α → α → Bool) (s : β → β → Bool) (l l' : List α)
+theorem map_merge {f : α → β} {r : α → α → Bool} {s : β → β → Bool} {l l' : List α}
     (hl : ∀ a ∈ l, ∀ b ∈ l', r a b = s (f a) (f b)) :
     (l.merge l' r).map f = (l.map f).merge (l'.map f) s := by
   match l, l' with
   | [], x' => simp
   | x, [] => simp
   | x :: xs, x' :: xs' =>
-    simp_rw [List.forall_mem_cons, forall_and] at hl
-    simp_rw [List.map, List.cons_merge_cons]
+    simp only [List.forall_mem_cons] at hl
+    simp only [forall_and] at hl
+    simp only [List.map, List.cons_merge_cons]
     rw [← hl.1.1]
     split
-    · rw [List.map, map_merge _ r s, List.map]
-      simp_rw [List.forall_mem_cons, forall_and]
+    · rw [List.map, map_merge, List.map]
+      simp only [List.forall_mem_cons, forall_and]
       exact ⟨hl.2.1, hl.2.2⟩
-    · rw [List.map, map_merge _ r s, List.map]
-      simp_rw [List.forall_mem_cons]
+    · rw [List.map, map_merge, List.map]
+      simp only [List.forall_mem_cons]
       exact ⟨hl.1.2, hl.2.2⟩
 
-theorem map_mergeSort' (f : α → β) (l : List α) (hl : ∀ a ∈ l, ∀ b ∈ l, a ≼ b ↔ f a ≼ f b) :
-    (l.mergeSort' r).map f = (l.map f).mergeSort' s :=
+theorem map_mergeSort {r : α → α → Bool} {s : β → β → Bool} {f : α → β} {l : List α}
+    (hl : ∀ a ∈ l, ∀ b ∈ l, r a b = s (f a) (f b)) :
+    (l.mergeSort r).map f = (l.map f).mergeSort s :=
   match l with
   | [] => by simp
   | [x] => by simp
   | a :: b :: l => by
-    simp_rw [← mem_split_iff (l := a :: b :: l), or_imp, forall_and] at hl
-    set l₁ := (split (a :: b :: l)).1
-    set l₂ := (split (a :: b :: l)).2
-    have e : split (a :: b :: l) = (l₁, l₂) := rfl
-    have fe : split (f a :: f b :: l.map f) = (l₁.map f, l₂.map f) := by
-      rw [← map, ← map, map_split, e]
-    have := length_split_fst_le l
-    have := length_split_snd_le l
-    simp_rw [List.map]
-    rw [List.mergeSort'_cons_cons _ e, List.mergeSort'_cons_cons _ fe,
-      map_merge _ (r · ·) (s · ·), map_mergeSort' _ l₁ hl.1.1, map_mergeSort' _ l₂ hl.2.2]
-    simp_rw [mem_mergeSort', decide_eq_decide]
-    exact hl.1.2
+    simp only [mergeSort, splitInTwo_fst, splitInTwo_snd, map_cons]
+    rw [map_merge (fun a am b bm => hl a (mem_of_mem_take (by simpa using am))
+      b (mem_of_mem_drop (by simpa using bm)))]
+    rw [map_mergeSort (s := s) (fun a am b bm => hl a (mem_of_mem_take (by simpa using am))
+      b (mem_of_mem_take (by simpa using bm)))]
+    rw [map_mergeSort (s := s) (fun a am b bm => hl a (mem_of_mem_drop (by simpa using am))
+      b (mem_of_mem_drop (by simpa using bm)))]
+    rw [map_take, map_drop]
+    simp
   termination_by length l
 
 end MergeSort
 
 end sort
 
--- try them out!
---#eval insertionSort (fun m n : ℕ => m ≤ n) [5, 27, 221, 95, 17, 43, 7, 2, 98, 567, 23, 12]
```

Comment:
I thought these comments were a cute demonstration of computability, I wouldn't want to see them go.

Reply:
I disagree?

Reply:
Can we turn this into an `example` instead?
We have more of such examples in mathlib...

Reply:
```
unseal merge mergeSort in
example :
    mergeSort [5, 27, 221, 95, 17, 43, 7, 2, 98, 567, 23, 12] (fun m n => m / 10 ≤ n / 10) =
      [5, 7, 2, 17, 12, 27, 23, 43, 95, 98, 221, 567] := rfl
```      
One needs some `unseal`ing here to run the example by `rfl`.